doc/tools/docgram/README.md

```diff
@@ -0,0 +1,199 @@
+`prodn.edit_mlg` and generates `prodnGrammar`, representing each production as
+a Sphinx prodn display.  Differently-edited grammars are used because prodn
+can naturally represent `LIST1 x SEP ','` whereas that is awkward for productionlists.
+
+6. TBA: Update the `tacn`, `cmd`, `tacv` and `cmdv` constructs in the `.rst` files.
+We still need to find a reasonable (i.e. not too tedious) way to associate each of
+these constructs with the production it describes.  For now, they must be updated by hand.
+
+How to use the tool
+-------------------
+`make doc_gram` updates `fullGrammar`.
+`make doc_gram_verify` verifies that `fullGrammar` is consistent with the `.mlg` files
+`make doc_gram_rsts` updates the `*Grammar` and `.rst` files.
+
+Changes to `fullGrammar`, `orderedGrammar` and the `.rst`s should be checked in to git.
+The other `*Grammar` files should not.
+
+Note: doc_grammar examines the current directory path so it updates the
+git-registered files rather than, say, the copy that Dune makes.  For example, if
+the current directory path is ".../coq/_build/default", then it prepends "../../"
+to the default path "doc/tools/docgram/fullGrammar".  (The tool looks for the "coq"
+directory in the current directory path.)
+
+Command line arguments
+----------------------
+The executable takes a list of `.mlg` and `.rst` files as arguments (in any order).
+Specifying the `-verify` command line argument avoids updating any of the files,
+but verifies that the current files are consistent.  This setting is meant for
+use in CI; it will be up to each developer to include the changes to `*Grammar` and
+the `.rst` files in their PRs when they've changed the grammar.
+
+Other command line arguments:
+`-check-tacs` reports on differences in tactics between the doc and the grammar
+`-check-cmds` reports on differences in commands between the doc and the grammar
+`-short` limits processing to updating/verifying only the fullGrammar file
+`-verbose` prints more messages about the grammar
+`-verify` described above
+
+Grammar editing scripts
+-----------------------
+
+The grammar editing scripts `*.edit_mlg` are similar in format to `.mlg` files stripped
+of all OCaml features.  This is an easy way to include productions to match or add without
+writing another parser.  The "DOC_GRAMMAR" token at the beginning of each file
+signals the use of streamlined syntax.
+
+Each edit file has a series of items in the form of productions.  Items are applied
+in the order they appear.  There are two types of editing operations:
+
+* Global edits - edit rules that apply to the entire grammar in a single operation.
+These are identified by using specific reserved names as the non-terminal name.
+
+* Local edits - edit rules that apply to the productions of a single non-terminal.
+If the non-terminal name isn't reserved, the rule is a local edit.  Local edits
+that begin with a different set of reserved names edit existing productions.  For
```

Comment:
This sentence is unclear. It's not the local edit itself which starts with a reserved name, it is a production within the local edit.